src/clj_icat_direct/queries.clj

```diff
@@ -198,8 +198,10 @@
                d.data_size                       AS data_size,
                d.create_ts                       AS create_ts,
                d.modify_ts                       AS modify_ts,
-               MAX(a.access_type_id)             AS access_type_id
-          FROM " objs-cte " AS d
+               MAX(a.access_type_id)             AS access_type_id"
+       (if include-count? ",
+               COUNT(*) OVER () AS total_count" "")
```

Comment:
Here and below, this is behind a flag because in the listing that includes both files and folders, we don't want this to be inside the `UNION`, it needs to be outside, because `UNION` is processed after window functions, so if we did it inside, the "total count" would show the total count of files for each file and the total count of folders for each folder, but the real overall total count nowhere. If we conclude at some point we want the separate counts (or I've neglected to notice we already do), changes can be made.

Reply:
We only need an overall count as far as I'm aware.